### PR TITLE
`Iris`: Allow only admins to change certain settings

### PR DESCRIPTION
**Pull request by Hialus**
> [!NOTE]
> Only deploy on TS9

#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

We want to prevent just any instructor from being able to enable Iris.

### Description

This PR adds a quick fix for this. This is only temporary until the settings system is reworked.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Admin (your TUM account)
- TS9

1. Log in to Artemis as Instructor
2. Navigate to a course management page
3. Open the Iris settings
4. Change the enabled status & the preferred model
5. Save
6. The values should be reverted
7. Go to an exercise management page
8. Do the same with the same result
9. Log in to Artemis as Admin
10. Repeat 2-8, but this time the settings actually save

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Unchanged

### CODE REVIEW

**Review by Strohgelaender — DISMISSED** [review]

**Review by MichaelOwenDyer — APPROVED** [review]
Good

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java [hunk 1]
```diff
@@ -379,8 +383,10 @@ private IrisSubSettings copyIrisSubSettings(IrisSubSettings target, IrisSubSetti
         if (target == null || source == null) {
             return source;
         }
-        target.setEnabled(source.isEnabled());
-        target.setPreferredModel(source.getPreferredModel());
+        if (authCheckService.isAdmin()) {
+            target.setEnabled(source.isEnabled());
+            target.setPreferredModel(source.getPreferredModel());
+        }
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:389]: Should we throw an exception when the user tries to change these values but does not have the rights to do so?

Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:389]: I don't think this is needed here. The UI will automatically switch back to the old values after saving, so this should indicate it to the user.
This is only a temporary and short-term fix, as I'm already working on reworking the settings system. As part of this rework, I will also handle this case properly.

However I now also disabled the fields in the UI if the user is not an admin, so this case can't even really happen.

**Review by JohannesStoehr — APPROVED** [review]
Approve code

**Review by DominikRemo — CHANGES_REQUESTED** [review]
Two tests in _IrisSettingsIntegrationTest_ fail on Bamboo and GitHub Actions

**Review by RY997 — DISMISSED** [review]
Code LGTM

**Review by MichaelOwenDyer — DISMISSED** [review]
LGTM 👍🏽

**Review by DominikRemo — APPROVED** [review]
Thank you for fixing the tests

**Review by MichaelOwenDyer — APPROVED** [review]
Reapprove

**Pull request merged**